docs/www/configuration.md

```diff
@@ -12,102 +12,406 @@ The `redpanda` section contains all the runtime configuration, such as the
+    
+  # Unique id identifying the node in the cluster.
+  # Required.
+  node_id: 1
+   
+  # Skips most of the checks performed at startup, not recomended for production use.
+  # Default: false
+  developer_mode: false
+
+  # The log segment size in bytes.
+  # Default: 1GiB
+  log_segment_size: 1073741824
+
+  # The compacted log segment size in bytes.
+  # Default: 256MiB
+  compacted_log_segment_size: 268435456
+
+  # The maximum compacted log segment size in bytes. The compaction process will
+  # attempt to combine segments to achieve a higher compaction rate. This maximum
+  # size will control how large a segment may become during this process.
+  # Default: 5 GiB
+  max_compacted_log_segment_size: 5368709120
+    
+  # The IP address and port for the internal RPC server.
+  # Default: 127.0.0.0:33145
+  rpc_server:
+    address: "0.0.0.0"
+    port: 33145
+   
+  # Multiple listeners are also supported as per KIP-103.
+  # The names must match those in advertised_kafka_api
+  kafka_api:
+  - address: "0.0.0.0"
+    name: internal
+    port: 9092
+  - address: "0.0.0.0"
+    name: external
+    port: 9093
+
+# TLS configuration for the RPC server.
+  # Default: null
+  rpc_server_tls:
+    # Whether to enable TLS for the RPC server.
+    enabled: false
+    # Require client authentication
+    require_client_auth: false
+    # The path to the server certificate PEM file.
+    cert_file: ""
+    # The path to the server key PEM file
+    key_file: ""
+    # The path to the truststore PEM file. Only required if client authentication
+    # is enabled.
+    truststore_file: ""
+  
+  # TLS configuration for the Kafka API.
+  # Default: null
+  kafka_api_tls:
+    # Whether to enable TLS for the Kafka API.
+    enabled: false
+    # Require client authentication
+    require_client_auth: false
+    # The path to the server certificate PEM file.
+    cert_file: ""
+    # The path to the server key PEM file
+    key_file: ""
+    # The path to the truststore PEM file. Only required if client authentication
+    # is enabled.
+    truststore_file: ""
+  
+  # Address and port of admin server.
+  # Default: 127.0.0.1:9644
+  admin:
+    address: "0.0.0.0"
+    port: 9644
+  
+  # TLS configuration for the admin server.
+  # Default: null
+  admin_api_tls:
+    # Whether to enable TLS for the admin server.
+    enabled: false
+    # Require client authentication
+    require_client_auth: false
+    # The path to the server certificate PEM file.
+    cert_file: ""
+    # The path to the server key PEM file
+    key_file: ""
+    # The path to the truststore PEM file. Only required if client authentication
+    # is enabled.
+    truststore_file: ""
+
+  # Address of RPC endpoint published to other cluster members.
+  # Default: 0.0.0.0:33145
+  advertised_rpc_api:
+    address: "0.0.0.0"
+    port: 33145
+
+  # Multiple listeners are also supported as per KIP-103.
+  # The names must match those in kafka_api
+  advertised_kafka_api:
```

Comment:
This is a long way from `kafka_api`, I think it would be nicer if it was colocated. Perhaps after `kafka_api_tls`